.github/workflows/ci.yml

```diff
@@ -0,0 +1,402 @@
+      - name: Upload lcov report
+        uses: actions/upload-artifact@v4
+        with:
+          name: report-coverage
+          path: lcov.info
+      - name: Upload coverage to Coveralls
+        uses: coverallsapp/github-action@v2.3.6
+        with:
+          file: lcov.info
+          format: lcov
+          github-token: ${{ github.token }}
+      - name: Upload test network artifacts
+        if: failure() && (steps.test.outcome == 'failure')
+        uses: actions/upload-artifact@v4
+        with:
+          name: test_network_runs
+          path: ${{ env.TEST_NETWORK_TMP_DIR }}/irohad_test_network_*
+          retention-days: 3
+
+  pytests:
+    needs: pre_build
+    runs-on: ubuntu-latest
+    env:
+      IROHA_CONTAINER: "defaults-irohad0-1"
+      BIN_IROHAD: bins/irohad
+      BIN_IROHA: bins/iroha
+      BIN_KAGAMI: bins/kagami
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.11.12
+      - uses: abatilo/actions-poetry@v4
+      - name: Download WAMSs
+        uses: actions/download-artifact@v4
+        with:
+          name: wasm
+          path: ${{ env.WASM_TARGET_DIR }}
+      - name: Move wasm libs
+        run: |
+          mv ${{ env.WASM_TARGET_DIR }}/libs ${{ env.DEFAULTS_DIR }}/libs
+          mv ${{ env.DEFAULTS_DIR }}/libs/default_executor.wasm ${{ env.DEFAULTS_DIR }}/executor.wasm
+      - name: Download binaries
+        uses: actions/download-artifact@v4
+        with:
+          name: bins
+          path: bins
+      - name: Make binaries executable
+        run: chmod +x -R bins
+      - name: Install test_env deps globally
+        run: pip install tomli-w
+      - name: Run test env
+        run: ./scripts/test_env.py setup
+      - name: Install Torii pytest dependencies
+        working-directory: pytests/iroha_torii_tests
+        run: poetry install --no-root
+      - name: Run Torii pytests
+        working-directory: pytests/iroha_torii_tests
+        run: poetry run pytest
+      - name: Install client pytest dependencies
+        working-directory: pytests/iroha_cli_tests
+        run: poetry install --no-root
+      - name: Run client pytests
+        uses: nick-fields/retry@v3
+        env:
+          # created by test_env.py
+          TMP_DIR: ../../test
+          IROHA_CLI_BINARY: iroha
+          IROHA_CLI_CONFIG: client.toml
+        with:
+          timeout_minutes: 10
+          max_attempts: 5
+          command: |
+            cd pytests/iroha_cli_tests
+            poetry run pytest
+          on_retry_command: |
+            ./scripts/test_env.py cleanup
+            ./scripts/test_env.py setup
+      - name: Terminate test network
+        run: ./scripts/test_env.py cleanup
+
+  # Run the job to check that the docker containers are properly buildable
+  pr_generator_build:
+    # Job will only execute if the head of the pull request is a branch for PR-generator case
```

Comment:
Is this job still being used @BAStos525 ?

Reply:
If you don't need it at all, we can remove then

Reply:
Removed